src/main/kotlin/com/openlattice/transporter/processors/TransporterPropagateDataEntryProcessor.kt

```diff
@@ -0,0 +1,109 @@
+package com.openlattice.transporter.processors
+
+import com.hazelcast.core.Offloadable
+import com.openlattice.edm.EntitySet
+import com.openlattice.postgres.PostgresArrays
+import com.openlattice.rhizome.hazelcast.entryprocessors.AbstractReadOnlyRhizomeEntryProcessor
+import com.openlattice.transporter.hasModifiedData
+import com.openlattice.transporter.tableName
+import com.openlattice.transporter.types.TransporterColumnSet
+import com.openlattice.transporter.types.TransporterDatastore
+import com.openlattice.transporter.types.TransporterDependent
+import com.openlattice.transporter.types.transporterNamespace
+import com.openlattice.transporter.updateIdsForEntitySets
+import com.openlattice.transporter.updateOneBatchForProperty
+import io.prometheus.client.Counter
+import org.slf4j.LoggerFactory
+import java.util.*
+
+class TransporterPropagateDataEntryProcessor(val entitySets: Set<EntitySet>, val partitions: Collection<Int>):
+        AbstractReadOnlyRhizomeEntryProcessor<UUID, TransporterColumnSet, Void?>(),
+        Offloadable
+{
+    companion object {
+        private val logger = LoggerFactory.getLogger(TransporterPropagateDataEntryProcessor::class.java)
+        val idCounter: Counter = Counter.build()
+                .namespace(transporterNamespace)
+                .name("inserted_ids")
+                .help("Rows updated from the ids table in the transporter database")
+                .register()
+        val valueCounter: Counter = Counter.build()
+                .namespace(transporterNamespace)
+                .name("updated_values")
+                .help("Rows updated in the transporter database")
+                .register()
+        val errorCounter: Counter = Counter.build()
+                .namespace(transporterNamespace)
+                .name("errors")
+                .help("Errors occurred during copies to the transporter database")
+                .register()
+    }
+
+    @Transient
+    private lateinit var data: TransporterDatastore
+
+    override fun process(entry: Map.Entry<UUID, TransporterColumnSet>): Void? {
+        check(::data.isInitialized) { TransporterDependent.NOT_INITIALIZED }
+        transport(entry)
+        return null
+    }
+
+    private fun transport(entry: Map.Entry<UUID, TransporterColumnSet>) {
+        if (entitySets.isEmpty() || partitions.isEmpty()) {
+            return
+        }
+        val tableName = tableName(entry.key)
+        entitySets.filter { it.isLinking }.forEach {
```

Comment:
Is this extra iteration only to log filtered linking entity sets?

Reply:
Yes. We can skip all the internal islinking checks here if we trust caller paths.